pkgs/applications/misc/st/default.nix

```diff
@@ -4,22 +4,20 @@
 with stdenv.lib;
 
 stdenv.mkDerivation rec {
-  name = "st-0.6";
+  name = "st";
+  version = "0.7";
   
   src = fetchurl {
-    url = "http://dl.suckless.org/st/${name}.tar.gz";
-    sha256 = "0avsfc1qp8zvshsfjwwrkvk411jlqy58z225bsdhjkl1qc40qcc5";
+    url = "http://dl.suckless.org/st/${name}-${version}.tar.gz";
+    sha1 = "358bca5bfc6ee85352dcd4f45f13f78474e5de42";
```

Comment:
Ah, sorry I missed that upstream provides sha1 sums. Then I suppose it is fine to reuse that here. In general though sha256 is preferred within Nixpkgs.

Personally I usually wouldn't put the upstream hash code but instead verify the upstream hash or GPG signature and then calculate a sha256 hash to put into Nixpkgs.